components/motor/client_test.go

```diff
@@ -171,31 +170,39 @@ func TestClient(t *testing.T) {
 	t.Run("client tests for failing motor", func(t *testing.T) {
 		err := failingMotorClient.GoTo(context.Background(), 42.0, 42.0, nil)
 		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errGoToFailed.Error())
 
 		err = failingMotorClient.ResetZeroPosition(context.Background(), 0.5, nil)
 		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errResetZeroFailed.Error())
 
 		pos, err := failingMotorClient.Position(context.Background(), nil)
 		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errPositionUnavailable.Error())
 		test.That(t, pos, test.ShouldEqual, 0.0)
 
 		err = failingMotorClient.SetPower(context.Background(), 42.0, nil)
 		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errSetPowerFailed.Error())
 
 		err = failingMotorClient.GoFor(context.Background(), 42.0, 42.0, nil)
 		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errGoForFailed.Error())
 
-		features, err := failingMotorClient.Properties(context.Background(), nil)
-		test.That(t, features, test.ShouldBeNil)
+		properties, err := failingMotorClient.Properties(context.Background(), nil)
+		test.That(t, properties.PositionReporting, test.ShouldBeFalse)
 		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errPropertiesNotFound.Error())
```

Comment:
are you able to use ShouldBeError here?

Reply:
I tried to but the tests were failing because the actual error contains a lot more into than the simple error I created. it seems like the best way to do it is compare the `error.Error()` strings